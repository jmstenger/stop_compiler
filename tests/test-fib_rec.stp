/*
 * test-fib_rec.stp
 * ====================
 */

def fib = (i:Int):Int {
	if (i == 0) {
		return 1;
	} else if (i == 1) {
		return 1;
	} else {
		return fib(i-1) + fib (i-2);
	}
}

def main = ():Int {
	printf("%d\n", fib(10));
	printf("%d\n", fib(9));
	printf("%d\n", fib(8));
	return 0;
}
