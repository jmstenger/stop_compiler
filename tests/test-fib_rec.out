89
55
34
